sips/sip-temp-derivation.md

```diff
@@ -0,0 +1,73 @@
+| SIP-Number          | <Leave this blank; it will be assigned by a SIP Editor> |
+| ---:                | :--- |
+| Title               | BLS-12381 Encryption Key Storage in Wallet |
+| Description         | Specify the key derivation path for the BLS-12381 Encryption Key for wallet master private key. |
+| Author              | Joy Wang (@joyqvq, joy@mystenlabs.com)|
+| Editor              | <Leave this blank; it will be assigned by a SIP Editor> |
+| Type                | Standard |
+| Category            | Wallet |
+| Created             | 2024-05-09 |
+| Comments-URI        | <Leave this blank; it will be assigned by a SIP Editor> |
+| Status              | <Leave this blank; it will be assigned by a SIP Editor> |
+| Requires            | <Optional; SIP number(s), comma separated> |
+
+## Abstract
+
+An increasing number of applications on Sui demand users to manage an encryption key on the client side. To address this need, we propose this standard and specification for deriving encryption keys from the master private key in the wallet. 
+
+The integration of encryption keys into wallets will expedite the adoption of numerous practical blockchain applications that rely on off-chain encryption and decryption. 
+
+## Motivation
+
+The standard is motivated by few potential applications such as encrypted NFT that requires the user to manage an encryption key in client side wallets. There are already many designs for applications that require encryption and decryption of on-chain data. 
+
+## Specification
+
+We leverage the master private key already stored in the wallet (either in Chrome Extension or mobile device) by defining a new derivation path. 
+
+We extend the specification for key derivation path to include a new unique derivation path in addition to all the signing keys for different signature schemes. This is domain separated for BLS12381 scheme and for encryption use only. 
+
+| Key scheme  | Use |  Derivation path  |
+|---|---|---|
+| Ed25519 |  Signature | m/44'/784'/{account}'/{change}'/{address}' |
+| ECDSA Secp256k1  | Signature | m/54'/784'/{account}'/{change}/{address} |
+| ECDSA Secp256r1  | Signature |  m/74'/784'/{account}'/{change}/{address} |
+| BLS12381  | Encryption | m/94'/784'/{account}'/{change}'/{address}' |
+
+We choose to use `94` on the second level to avoid any collisions with key derivation path for other blockchains and use cases. 
+
+To generate multiple BLS-12381 encryption key from the same master private key, it is recommended to increment the `account` level (e.g. `m/94'/784'/0'/0'/0'`, `m/94'/784'/1'/0'/0'`, `m/94'/784'/2'/0'/0'`, etc). Note that the derivation path for the first three rows are already adopted and supported in Sui. The key derivation path is an extension to [SLIP-0010](https://github.com/satoshilabs/slips/blob/master/slip-0010.md). 
```

Comment:
We should update reference to SLIP10, cause BLS12-381 is using longer keys, 48 bytes, and Slip10 flows are only working for 32 bytes

Reply:
here we derive the seed for the BLS keygen so it converts from 32 to 64 bytes. alternatively we can use this spec: https://eips.ethereum.org/EIPS/eip-2333#hkdf_mod_r for hash to element.